src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt

```diff
@@ -7,13 +7,28 @@ import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.HdcLicence
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.hdc.HdcApiClient
 import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.hdc.HdcLicenceData
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.prison.PrisonApiClient
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceKind
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceKind.HDC
+import java.time.LocalDate
 
 @Service
 class HdcService(
   private val hdcApiClient: HdcApiClient,
+  private val prisonApiClient: PrisonApiClient,
   private val licenceRepository: LicenceRepository,
 ) {
 
+  fun <T> getHdcStatus(
+    records: Collection<T>,
+    bookingIdGetter: (T) -> Long?,
+    hdcedGetter: (T) -> LocalDate?,
+  ): HdcStatuses {
```

Comment:
keeping generic to not tie to a specific type
(Should work with licences / search results etc..)